### DISCUSSION TITLE
Compiling with system openssl

### DISCUSSION THREAD

**Discussion by Gigas002** [microsoft/msquic#4019] [Q&A] [answered]
Is it possible to compile `msquic` with system-wide installation of `openssl` instead of submodule? The `openssl`'s release `3.2.0` seems to [have added](https://github.com/openssl/openssl/blob/openssl-3.2.0/README-QUIC.md) the ability to work with `quic` protocol so I thought it might be usable, please correct me if I'm wrong

**Comment by nibanks** [comment, marked as answer]
No, we don't yet support openssl v3.2 because it added _some_ QUIC support _inside_ the project, instead of exposing the functionality that QUIC needs to be built _on top_ of it. In addition, these changes have caused a lot of name conflicts with the quictls fork that we used to build against.

Bottom line, it has been a struggle to work with openssl as the TLS provider for QUIC, and it's only getting more difficult.

**Reply by Gigas002** [reply]
Ah, I see. Thanks for explanation

**Comment by samitanwer** [comment]
Is there an update on the support for OSSL ver. >v3.2? I am particularly interested in support for v3.3. 
It would be nice to know where one can track progress, maybe [quic-on-3.3-dirty](https://github.com/quictls/openssl/tree/wladd/quic-on-3.3-dirty) ?? But this branch has not been touched in the last 2 months.

**Reply by nibanks** [reply]
We have a draft PR to support it, but have some issues. Hopefully we can get those worked out soon.

**Reply by samitanwer** [reply]
Could you point me to the PR please?

**Reply by nibanks** [reply]
https://github.com/microsoft/msquic/pull/4274
